File: javalin/src/test/java/io/javalin/examples/HelloWorldSecure.java
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 *
 */

package io.javalin.examples;

import io.javalin.Javalin;
import org.eclipse.jetty.server.ServerConnector;
import org.eclipse.jetty.util.ssl.SslContextFactory;

public class HelloWorldSecure {

    // This is a very basic example, a better one can be found at:
    // https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/examples/embedded/src/main/java/org/eclipse/jetty/embedded/LikeJettyXml.java#L139-L163
    public static void main(String[] args) {
        Javalin.create(config -> {
            config.jetty.addConnector((server, httpConfiguration) -> {
                    ServerConnector sslConnector = new ServerConnector(server, getSslContextFactory());
                    sslConnector.setPort(443);
                    return sslConnector;
            });
            config.jetty.addConnector((server, httpConfiguration) -> {
                    ServerConnector connector = new ServerConnector(server);
                    connector.setPort(80);
                    return connector;
            });
            config.routes.get("/", ctx -> ctx.result("Hello World")); // valid endpoint for both connectors

        }).start();
    }

    private static SslContextFactory.Server getSslContextFactory() {
        SslContextFactory.Server sslContextFactory = new SslContextFactory.Server();
        sslContextFactory.setKeyStorePath(HelloWorldSecure.class.getResource("/keystore.jks").toExternalForm());
        sslContextFactory.setKeyStorePassword("password");
        return sslContextFactory;
    }

}

File: javalin/src/test/kotlin/io/javalin/examples/HelloWorldSecure.kt
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 */

package io.javalin.examples

import io.javalin.Javalin
import org.eclipse.jetty.server.ServerConnector
import org.eclipse.jetty.util.ssl.SslContextFactory

// This is a very basic example, a better one can be found at:
// https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/examples/embedded/src/main/java/org/eclipse/jetty/embedded/LikeJettyXml.java#L139-L163
fun main() {
    val app = Javalin.create {
        it.jetty.addConnector { server, _ ->
            val sslConnector = ServerConnector(server, sslContextFactory())
            sslConnector.port = 443
            sslConnector
        }
        it.jetty.addConnector { server, _ ->
            val httpConnector = ServerConnector(server)
            httpConnector.port = 80
            httpConnector
        }
    }.start()

    app.get("/") { it.result("Hello World") } // valid endpoint for both connectors
}

private fun sslContextFactory(): SslContextFactory.Server {
    val sslContextFactory = SslContextFactory.Server()
    sslContextFactory.keyStorePath = HelloWorldSecure::class.java.getResource("/keystore.jks")!!.toExternalForm()
    sslContextFactory.keyStorePassword = "localhost"
    return sslContextFactory
}
